OpenRA.Mods.Common/Widgets/RadarWidget.cs

```diff
@@ -137,15 +139,23 @@ void UpdateTerrainCell(CPos cell)
 				return;
 
 			var custom = world.Map.CustomTerrain[uv];
+			var tileset = world.Map.Rules.TileSet;
 			int leftColor, rightColor;
 			if (custom == byte.MaxValue)
 			{
-				var type = world.Map.Rules.TileSet.GetTileInfo(world.Map.Tiles[uv]);
+				var type = tileset.GetTileInfo(world.Map.Tiles[uv]);
 				leftColor = type != null ? type.LeftColor.ToArgb() : Color.Black.ToArgb();
 				rightColor = type != null ? type.RightColor.ToArgb() : Color.Black.ToArgb();
 			}
 			else
-				leftColor = rightColor = world.Map.Rules.TileSet[custom].Color.ToArgb();
+			{
+				// special case as resources are hidden under fog
+				var resource = resourceLayer.GetRenderedResource(cell);
+				if (resource != null)
```

Comment:
This can be simplified:

```csharp
if (resource != null)
    custom = tileset.GetTerrainIndex(resource.Info.TerrainType);

leftColor = rightColor = world.Map.Rules.TileSet[custom].Color.ToArgb();
```